ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationIT.scala

```diff
@@ -69,23 +70,25 @@ final class CommandDeduplicationIT(
         _.commands.deduplicationPeriod :=
           DeduplicationPeriod.DeduplicationDuration(deduplicationDuration.asProtobuf)
       )
-    val acceptedSubmissionId1 = newSubmissionId()
+    val firstAcceptedSubmissionId = newSubmissionId()
+    val firstSubmissionSendTime = Instant.now()
     for {
       // Submit command (first deduplication window)
       // Note: the second submit() in this block is deduplicated and thus rejected by the ledger API server,
       // only one submission is therefore sent to the ledger.
       response <- submitRequestAndAssertCompletionAccepted(
         ledger,
-        updateSubmissionId(request, acceptedSubmissionId1),
+        updateSubmissionId(request, firstAcceptedSubmissionId),
         party,
       )
-      _ <- submitRequestAndAssertDeduplication(
+      optCompletion <- submitRequestAndAssertDeduplication(
```

Comment:
I recommend to use a longer deduplication duration than 2s here. On a high-latency ledger implementation such as Canton on Besu, it may well be that it takes > 2s to execute a single command. In that case, the second submission would be accepted, i.e., the test becomes flaky.

Reply:
@andreaslochbihler-da, what value would you propose? I'm a bit concerned about this, as a few tests wait for the whole duration to resubmit.

Reply:
I recommend to use use different durations in different test cases. Specify long durations (at least minutes) where you expect an `ALREADY_EXISTS` and short durations where you expect the request to be accepted.

Reply:
@andreaslochbihler-da I would recommend that we increase the time factor in that case. Makes the test cases a lot easier to reason about if we keep things simple as to what deduplication duration we select

Reply:
It's a trade-off. It's your CI costs that will rise because wait times are longer than technically necessary. If that's less a concern than the complexity added by varying dedup period, then let's go for the time factor.

Reply:
@nicu-da, the time factor mechanism is in place, AFAICS. Do we need any changes in this case?

Reply:
I would not do any changes for now.